pilot/cmd/pilot-agent/main.go

```diff
@@ -196,6 +200,50 @@ var (
 
 			log.Infof("Monitored certs: %#v", certs)
 
+			if role.Type == model.Static && proxyConfig.CustomConfigFile == "" {
+				var upstreams []bootstrap.Upstream
+				var telemetry *bootstrap.Upstream
+				switch staticProfile {
+				case "telemetry":
+					telemetry = &bootstrap.Upstream{
+						ListenPort:   15004,
+						UpstreamPort: 9091,
+						GRPC:         true,
+						Auth:         proxyConfig.ControlPlaneAuthPolicy == meshconfig.AuthenticationPolicy_MUTUAL_TLS,
+						Service:      fmt.Sprintf("%s.%s", bootstrap.TelemetryAddress, role.Domain),
+						UID:          fmt.Sprintf("kubernetes://%s", role.ID),
+						Operation:    "Check",
+					}
+					upstreams = []bootstrap.Upstream{*telemetry}
+				case "policy":
+					upstreams = []bootstrap.Upstream{{
+						ListenPort:   15004,
+						UpstreamPort: 9091,
+						GRPC:         true,
+						Auth:         proxyConfig.ControlPlaneAuthPolicy == meshconfig.AuthenticationPolicy_MUTUAL_TLS,
+						Service:      fmt.Sprintf("%s.%s", bootstrap.PolicyAddress, role.Domain),
+						UID:          fmt.Sprintf("kubernetes://%s", role.ID),
+						Operation:    "Report",
+					}}
+				}
+
+				config, err := bootstrap.BuildBootstrap(upstreams, telemetry, proxyConfig.ZipkinAddress, proxyConfig.ProxyAdminPort)
+				if err != nil {
+					return err
+				}
+				marshaler := jsonpb.Marshaler{OrigName: true, Indent: "  "}
+				out, err := marshaler.MarshalToString(config)
```

Comment:
can we start using `yaml` for bootstrap ? It seems envoy uses yaml in examples, and it is arguably more readable.

Reply:
Sure. I'd actually prefer using `--config-yaml` option here that does not use files in-between, but it hasn't been merged yet.